modules/oe_whitelabel_search/src/Form/SearchForm.php

```diff
@@ -0,0 +1,116 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\oe_whitelabel_search\Form;
+
+use Drupal\Core\Form\FormBase;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\Language\LanguageManagerInterface;
+use Drupal\Core\Url;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Exposes the facets as a form.
+ */
+class SearchForm extends FormBase {
+
+  /**
+   * The language manager.
+   *
+   * @var \Drupal\Core\Language\LanguageInterface
+   */
+  protected $languageManager;
+
+  /**
+   * Constructs an instance of ListFacetsForm.
+   *
+   * @param \Drupal\Core\Language\LanguageManagerInterface $language_manager
+   *   The language manager.
+   */
+  public function __construct(LanguageManagerInterface $language_manager) {
+    $this->languageManager = $language_manager;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container) {
+    return new static(
+      $container->get('language_manager'),
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function buildForm(array $form, FormStateInterface $form_state, array $config = NULL): array {
+    $form_state->set('oe_whitelabel_search_config', $config);
+    $enable_autocomplete = $config['view_options']['enable_autocomplete'] ?? $config['enable_autocomplete'];
+    $form['#action'] = $config['form']['action'] ?? $config['form_action'];
+    isset($config['input']['name']) ? $parameter = \Drupal::request()->get($config['input']['name']) : $parameter = '';
+    isset($config['input']['classes']) ? $classes = $config['input']['classes'] : $classes = '';
+    isset($config['button']['classes']) ? $classesButton = $config['button']['classes'] : $classesButton = '';
+    if (empty($classes) && isset($config['input_classes'])) {
+      $classes = $config['input_classes'];
+    }
+    $form['search_input'] = [
+      '#type' => 'textfield',
+      '#attributes' => [
+        'placeholder' => $config['input']['placeholder'] ?? $config['input_placeholder'],
+        'class' => (array) $classes,
+      ],
+      '#default_value' => $parameter,
+      '#required' => TRUE,
+    ];
+    if ($enable_autocomplete) {
```

Comment:
`if (!$config['view_options']['enable_autocomplete']) {
      return $form;
    }`

and then put the rest of the code below, this way we add clarity